README.md

```diff
@@ -120,13 +132,27 @@ Current state | Next word | Next state
 --------------|-----------|-----------
     S0        |    the    |     S1
     S1        |   . . .   |   . . .
+    
 
 (hint: my version has 13 entries. Yours _might_ be different)
 
 ## A2.4
 
-«replace this with your answer»
-
+Current state | Next word | Next state
+--------------|-----------|-----------
+    S0        |    the    |     S1
+    S1        |    lazy   |     S2
+    S1        |    smelly |     S2
+    S1        |    ""     |     S2
+    S1        |    lazy   |     S1
+    S1        |    smelly |     S1
+    S2        |    cat    |     S3
+    S2        |    dog    |     S3
+    S3        |    ran    |     S4
+    S3        |    ate    |     S4
+    S4        |    slowly |     eoi
+    S4        |    noisily|     eoi
+    S4        |    ""     |     eoi
```

Comment:
This table doesn't really agree with your diagram. For example, you diagram has two transitions from S0, and this table only has one.

This table also has transitions marked with an empty string. That can't work, because the state machine would never know when to apply it. For example, it's in S1. The next word is "lazy." The state machine _could_ choose it. Or it could simply say "I'll match nothing and go to S2", breaking the parse.

# -3

Here's a version of the table for the diagram above

Current State | Next Word | Next State
------------- | --------- | ----------
S0            | "the"     | S1
S1            | "lazy"    | S1
S1            | "smelly"  | S1
S1            | "dog"     | S2
S1            | "cat"     | S2
S2            | "ate      | S3
S2            | "ran"     | S3
S3            | "slowly"  | S4
S3            | "noisily" | S4
S3            | EOI       | END
S4            | EOI       | END